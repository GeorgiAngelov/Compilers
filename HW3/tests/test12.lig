type a : [[int]];

type b : {head : a, rest : b};

type c : [b];

var x : a = [nil, nil, [1]];

var y : a = [nil, [2], nil];

var z : b = {head = [nil], rest = {head = [[1], nil], rest = nil}};

var w : c = [{head=nil, rest=nil}, {head=[[1]], rest={head=nil, rest=nil}}];
