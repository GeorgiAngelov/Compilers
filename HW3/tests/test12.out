(decls
   (decl-type:ok (id a) (type-array (type-array (type-int))))
   (decl-type:ok (id b) (type-struct (typed-id (id head) (type-id a)) (typed-id (id rest) (type-id b))))
   (decl-type:ok (id c) (type-array (type-id b)))
   (decl-var:ok (id x) (type-id a) (array-lit:[[int]] (exps (nil:nil) (nil:nil) (array-lit:[int] (exps (num:int 1))))))
   (decl-var:ok (id y) (type-id a) (array-lit:[[int]] (exps (nil:nil) (array-lit:[int] (exps (num:int 2))) (nil:nil))))
   (decl-var:ok (id z) (type-id b) (struct-lit:{head : [nil], rest : {head : [[int]], rest : nil}} (field-inits (field-init (id head) (array-lit:[nil] (exps (nil:nil)))) (field-init (id rest) (struct-lit:{head : [[int]], rest : nil} (field-inits (field-init (id head) (array-lit:[[int]] (exps (array-lit:[int] (exps (num:int 1))) (nil:nil)))) (field-init (id rest) (nil:nil))))))))
   (decl-var:ok (id w) (type-id c) (array-lit:[{head : [[int]], rest : {head : nil, rest : nil}}] (exps (struct-lit:{head : nil, rest : nil} (field-inits (field-init (id head) (nil:nil)) (field-init (id rest) (nil:nil)))) (struct-lit:{head : [[int]], rest : {head : nil, rest : nil}} (field-inits (field-init (id head) (array-lit:[[int]] (exps (array-lit:[int] (exps (num:int 1)))))) (field-init (id rest) (struct-lit:{head : nil, rest : nil} (field-inits (field-init (id head) (nil:nil)) (field-init (id rest) (nil:nil)))))))))))
Function main() not defined or has wrong type!
